### PR TITLE
Switch to ESLint flat conf and stricter rules

### PR DESCRIPTION
**Pull request by throwaway96**
I ported everything I could from the `.eslintrc.json` into `eslint.config.mjs`. I think I got everything meaningful. The result is pretty ugly, although that seems to be par for the course with flat configs.

It checks JS and TS files in `services`, JS files in `tools`, `webpack.config.js`, and `eslint.config.mjs` itself. It also now checks JSON files, at least for syntax errors. This new config format should make it relatively easy (😬) to add support for linting `frontend`.

There were a bunch of changes required to pass the new checks. The related changes for `elevate-service.ts` are in #173.

Some of the changes are not strictly necessary for passing the new checks. I am thinking about breaking these out into a separate commit/PR. My goal is to code defensively and provide users with enough feedback to determine the cause of an error. I'm hoping that when LG inevitably changes something in the future, users will get more useful error messages than `undefined is not a function` or `Cannot read property 'x' of undefined`.

This PR also updates two runtime dependencies. (Note that `dompurify` 3.0.2+ causes `enyo pack --production` to fail.)

I haven't tested this yet, but it's pretty big so I'll put it here now in case anyone wants to take a look at it.

### CODE REVIEW

File: services/adapter.ts [hunk 1]
```diff
@@ -8,21 +8,21 @@ import * as Bluebird from 'bluebird';
...
+if (typeof String.prototype.normalize !== 'function') {
   // eslint-disable-next-line no-extend-native
   String.prototype.normalize = function normalize() {
     return String(this);
   };
 }
 
 // Monkey-patch fetch Promise with Bluebird's.
-// @ts-ignore
+// @ts-expect-error
```
Comment (kitsuned) [inline services/adapter.ts:19]: `ts-expect-error` is used in unit tests (e.g. we want to test runtime type assertions).

i think `ts-ignore` is more relevant.

File: services/adapter.ts [hunk 2]
```diff
@@ -38,5 +38,8 @@ export const asyncWriteFile: (path: string, contents: string, options?: fs.Write
   fs.writeFile,
 );
 export const asyncChmod: (path: string, mode: fs.Mode) => Promise<void> = Bluebird.Promise.promisify(fs.chmod);
-export const asyncExists: (path: string) => Promise<boolean> = (path) => new Promise((resolve) => fs.exists(path, resolve));
+export const asyncExists: (path: string) => Promise<boolean> = (path) =>
+  new Promise((resolve) => {
+    fs.exists(path, resolve);
```
Comment (kitsuned) [inline services/adapter.ts:43]: `exists` is deprecated, we should use `stat`.

File: services/webos-service-remote/message.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ export default class Message {
     this.token = body.token();
     try {
       this.payload = JSON.parse(body.payload());
-    } catch (e) {
+    } catch (_e) {
```
Comment (kitsuned) [inline services/webos-service-remote/message.js:17]: `catch (_e) { ... }` -> `catch { ... }`

**Review by kitsuned — CHANGES_REQUESTED** [review]

File: services/adapter.ts (hunk 1 above)
Comment (throwaway96) [reply services/adapter.ts:19]: I like the reasoning behind the [`prefer-ts-expect-error`](https://typescript-eslint.io/rules/prefer-ts-expect-error/) rule.

File: services/adapter.ts (hunk 2 above)
Comment (throwaway96) [reply services/adapter.ts:43]: Addressed in #175.

File: services/adapter.ts (hunk 1 above)
Comment (kitsuned) [reply services/adapter.ts:19]: i won't be persistent; it's not *that* important.

**Review by kitsuned — APPROVED** [review]

**Pull request merged**